performance-tests/bench_2/worker/SetCftParametersAction.cpp

```diff
@@ -0,0 +1,141 @@
+#include "SetCftParametersAction.h"
+
+#include "MemFunHandler.h"
+
+namespace {
+
+uint32_t one_at_a_time_hash(const uint8_t* key, size_t length) {
+  size_t i = 0;
+  uint32_t hash = 0;
+  while (i != length) {
+    hash += key[i++];
+    hash += hash << 10;
+    hash ^= hash >> 6;
+  }
+  hash += hash << 3;
+  hash ^= hash >> 11;
+  hash += hash << 15;
+  return hash;
+}
+
+const ACE_Time_Value ZERO(0, 0);
+
+}
+
+namespace Bench {
+
+  SetCftParametersAction::SetCftParametersAction(ACE_Proactor& proactor)
+ : proactor_(proactor), started_(false), stopped_(false)
+ , set_period_(1, 0), max_count_(0), set_call_count_(0)
+{
+}
+
+bool SetCftParametersAction::init(const ActionConfig& config, ActionReport& report,
+  Builder::ReaderMap& readers, Builder::WriterMap& writers, const Builder::ContentFilteredTopicMap& cft_map)
+{
+
+  std::unique_lock<std::mutex> lock(mutex_);
+  Action::init(config, report, readers, writers, cft_map);
+
+  if (!content_filtered_topic_) {
```

Comment:
Instead of expecting the common Action::Init() to populate a content filtered topic name, let's check for a config parameter called "content_filtered_topic_name" (or something like that) here and then check the map. Alternatively, you could even have the action attach a reader and use get_topicdescription on the reader to get a handle to the CFT.